### PR TITLE
Added "when" method for validation at condition

### PR DESCRIPTION
**Pull request by thenotsoft**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  |

### CODE REVIEW

**Comment by samdark** [comment]
@thenotsoft would you please resolve conflicts?

**Comment by samdark** [comment]
@thenotsoft seems to fail tests.

File: src/Rule.php [hunk 1]
```diff
@@ -31,7 +32,10 @@ abstract class Rule
             return new Result();
         }
 
-        if ($this->skipOnError && $previousRulesErrored) {
+        if (
+          ($this->skipOnError && $previousRulesErrored) ||
+          ($this->when !== null && !call_user_func($this->when))
```
Comment (xepozz) [inline src/Rule.php]:
```suggestion
          (is_callable($this->when) && !call_user_func($this->when))
```

File: src/Rule.php [hunk 2]
```diff
@@ -16,6 +16,7 @@ abstract class Rule
     private ?string $translationLocale = null;
     private bool $skipOnEmpty = false;
     private bool $skipOnError = true;
+    private $when = null;
```
Comment (xepozz) [inline src/Rule.php]: Please add PHPDoc

File: tests/ValidatorTest.php [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 namespace Yiisoft\Validator\Tests;
 
+use Yiisoft\Validator\Rule;
```
Comment (xepozz) [inline tests/ValidatorTest.php]: I don't see the usings

**Comment by samdark** [comment]
Adjusted it and merged. Thanks!

**Pull request merged**